compiler/modules/front.syntax/parseTypeDesc.bal

```diff
@@ -69,8 +69,14 @@ function parsePostfixTypeDesc(Tokenizer tok) returns TypeDesc|err:Syntax {
         }
         else if tok.current() == "[" {
             check tok.advance();
+            Token? c = tok.current();
+            SingletonTypeDesc? length = ();
+            // todo: this only allow int literals to be within brackets, should be allow any numeric literal and then show appropriate error later?
```

Comment:
You've got to allow a reference to a const, which you have to resolve during resolveTypes.